File: frameworks/go/gorilla-mux/Implementation/json.go
package implementation

import (
	"net/http"

	"github.com/gorilla/mux"
)

// jsonRoutes serialise a payload the framework already holds, at three sizes, with
// encoding/json's encoder.
func jsonRoutes(r *mux.Router, p *Payloads) {
	r.HandleFunc("/json/small", func(w http.ResponseWriter, r *http.Request) { respond(w, http.StatusOK, &p.Small) }).Methods(http.MethodGet)

	r.HandleFunc("/json/medium", func(w http.ResponseWriter, r *http.Request) { respond(w, http.StatusOK, &p.Medium) }).Methods(http.MethodGet)

	r.HandleFunc("/json/large", func(w http.ResponseWriter, r *http.Request) { respond(w, http.StatusOK, &p.Large) }).Methods(http.MethodGet)
}

File: frameworks/rust/salvo/Implementation/routes/json.rs
use salvo::prelude::*;

use crate::Payloads;
use crate::answers::Serialised;

/// json: a payload the framework already holds, serialised at three sizes.
pub fn router(p: &'static Payloads) -> Router {
    Router::with_path("/json")
        .push(Router::with_path("small").get(Serialised(&p.small)))
        .push(Router::with_path("medium").get(Serialised(&p.medium)))
        .push(Router::with_path("large").get(Serialised(&p.large)))
}
